### ISSUE TITLE
Date web component with text inputs - Accessibility review

### ISSUE THREAD

**Issue by k80bowman** [department-of-veterans-affairs/vets-design-system-documentation#844]
## Issue Description
We need to conduct a full accessibility review of the experimental Date component with text inputs. The component can be viewed in Storybook here: https://design.va.gov/storybook/?path=/docs/components-va-memorable-date--default

## Tasks
- [ ] Full accessibility review of the component is completed
- [ ] Add descriptions of any issues found to this ticket
    - [ ] If any problems are found, add screen recordings and/or screenshots to comments on this ticket that show the problem.
- [ ] If necessary, discuss any accessibility issues found with the engineer who developed the component
- [ ] Re-review any fixes the developer makes to the component

## Acceptance Criteria
- [ ] Component has been reviewed for accessibility issues
- [ ] Any accessibility issues have been addressed

**Comment by k80bowman** [comment]
@SarahKay8 could you take a look at this one please?

**Comment by k80bowman** [comment]
@SarahKay8 I updated the Storybook link for this component above.

**Comment by TiffanyPender** [comment]
Including a few accessibility feedback notes on the memorable date component below. (Using the feedback format, which may be helpful in determining priority)

**The Positives**
- A legend is included with helpful hint text
- There is a label associated with each input field
- The field borders and associated text appear thicker/bolded when errors are present - the indication of an error is not reliant on color alone. 

----

**Additional Feedback**
**❗️ Must** for if the feedback must be applied:
- Ensure that there are no duplicate for/ID attributes on label/input field groupings
  - This may have been a minor miss in the demo version, but just wanted to note it here - <label> - “for” attribute AND <input> - “id” attribute should be different for each field group
- Add in the aria-invalid="true" attribute when input triggers errors

**⚠️ Should** if the feedback is best practice: 
- Remove the aria-live=”polite” attribute included after each input field. These are in conflict with role=”alert” and are never announced. It may be helpful to consider if the error information should be available to all users, not just screen reader users.

**✔️ Consider** for suggestions/enhancements:
- The attribute aria-describedby="dateHint" is included on all <va-text-input> elements, but is not read by the screen reader for each group that includes the label + input field. I would recommend either of the following:
  -  Removing the aria-describedby attributes, and instead, including text for screen readers within the <label>, specific to each input field; 
  -  Including the aria-describedby attributes on each input element with helpful text specific to each input field
- Consider adding context to the error message to let users know which exact fields are producing an error

**Comment by TiffanyPender** [comment]
For the duplicate IDs on the input elements mentioned in the previous comments, this does not seem to be an accessibility blocker. As @bkjohnson mentioned, the IDs do not appear in the same DOM. In screen reader testing, as well as looking at the form control rotor, it lists each field separately with the appropriate label. And automated testing with Axe DevTools does not list this as an issue. Maybe something to keep an eye on in the future, but not currently an accessibility blocker.

**Comment by bkjohnson** [comment]
I've merged two PRs which address some of the problems here:

- https://github.com/department-of-veterans-affairs/component-library/pull/485
- https://github.com/department-of-veterans-affairs/component-library/pull/484

All that remains is "Add in the aria-invalid="true" attribute when input triggers errors" from the **Must** category, and both of the **Consider** items

**Comment by TiffanyPender** [comment]
The `aria-invalid="true"` should be placed on each individual `<input>` element, when an error is present. This will help to notify screen readers that an error is present for specific fields.

**Comment by bkjohnson** [comment]
The date component doesn't have any `<input>` elements as direct children. I was planning on taking that approach for the smaller components since we currently don't do that at all, but since [having `aria-describedby` on the `<input>` inside of `<va-text-input>` wasn't enough to satisfy the error messaging for `<va-date>`](https://github.com/department-of-veterans-affairs/va.gov-team/issues/39727#issuecomment-1182141099), I assumed that the same situation would be true with `aria-invalid`.

**Comment by bkjohnson** [comment]
The last of the "Must/Should" feedback was addressed in the previous PR. I'll open a new ticket to track the "Consider" items.
